src/neural-network.js

```diff
@@ -366,6 +392,22 @@ export default class NeuralNetwork {
     return true;
   }
 
+
+  /**
+   *
+   * @param options
+   * @private
+   */
+  _validateTrainingOptions(options) {
+    Object.keys(options).forEach(key => {
+      if (NeuralNetwork.trainingValidations[key] && !NeuralNetwork.trainingValidations[key](options[key])) {
+        let message = `[${key}, ${options[key]}] is out of noraml range`;
+        if (this._isStrict) throw new Error (message);
```

Comment:
If these settings are wrong, this will break training, right? Why not just always enforce `isStrict`?

Reply:
I thought it would let the class stay more flexible to allow the validation to be shut off (in the odd case there is some weird option that they legitimately want to allow).  I kinda wrestled with this, it's that balance between providing structure and allowing innovation/creativity.  I landed on this due to the goal of the ticket, to help those who are new figure out things that might be going wrong.  Enforcing unexpected behavior I felt didn't fall within that (even though I can't think of a reason why they would want to do that).